stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChannelData.kt

```diff
@@ -14,28 +14,30 @@ import java.util.Date
 public data class ChannelData(
     var type: String,
     var channelId: String,
-    val cooldown: Int = 0
-) {
-    var cid: String = "%s:%s".format(type, channelId)
+    var cid: String = "%s:%s".format(type, channelId),
 
     /** created by user */
-    var createdBy: User = User()
+    var createdBy: User = User(),
+
+    var cooldown: Int = 0,
 
     /** if the channel is frozen or not (new messages wont be allowed) */
-    var frozen: Boolean = false
+    var frozen: Boolean = false,
 
     /** when the channel was created */
-    var createdAt: Date? = null
+    var createdAt: Date? = null,
     /** when the channel was updated */
-    var updatedAt: Date? = null
+    var updatedAt: Date? = null,
     /** when the channel was deleted */
-    var deletedAt: Date? = null
+    var deletedAt: Date? = null,
     /** all the custom data provided for this channel */
     var extraData: MutableMap<String, Any> = mutableMapOf()
+) {
```

Comment:
this data class had the important fields specified in the body. the comparison operators don't work on the body fields (only the arguments). because the objects compared as equal, and stateflow filters out data that's the same, updates on channel data didn't work.